com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/util/ComponentsCounter.java

```diff
@@ -307,21 +311,26 @@ public static void copyDataToBuilder(Storage storage, Project project, Collectio
         }
         Map<String, Integer> components = storage.get();
         HashMap<String, Integer> mergedComponents = new HashMap<>();
+        boolean hasDynamicValue = false;
         for (Map.Entry<String, Integer> entry : components.entrySet()) {
             // different input component names may have the same output name
             // for example wav and sound both are soundc
             String name = project.replaceExt("." + entry.getKey()).substring(1);
+            Integer value = entry.getValue();
             if (mergedComponents.containsKey(name)) {
                 Integer mergedValue = mergedComponents.get(name);
-                Integer value = entry.getValue();
                 if (mergedValue.equals(DYNAMIC_VALUE) || value.equals(DYNAMIC_VALUE)) {
                     mergedComponents.put(name, DYNAMIC_VALUE);
                 } else {
                     mergedComponents.put(name, mergedValue + value);
                 }
             } else {
-                mergedComponents.put(name, entry.getValue());
+                mergedComponents.put(name, value);
             }
+            hasDynamicValue |= isFactoryType(name, true);
```

Comment:
If a collection has at least one factory, `goc` should be `DYNAMIC_VALUE` - used from `game.project`.